oak_functions/loader/src/lookup.rs

```diff
@@ -109,7 +109,7 @@ where
         Ok(extension_result)
     }
 
-    fn get_metadata(&self) -> (String, wasmi::Signature) {
+    fn get_metadata(&self) -> (String, wasmi::Signature, ExtensionHandle) {
```

Comment:
+1

Reply:
Thank you, that's true! Instead of adding a struct, I added a new `get_handle` function to the trait. I believe we'll remove `get_metadata` eventually, so that makes more sense.